presto-druid/src/main/java/com/facebook/presto/druid/DruidBrokerPageSource.java

```diff
@@ -123,17 +121,10 @@ public Page getNextPage()
                 }
                 else {
                     JsonNode rootNode = OBJECT_MAPPER.readTree(readLine);
-                    if (!(rootNode instanceof ArrayNode)) {
-                        if (rootNode instanceof ObjectNode) {
-                            throw new PrestoException(DRUID_BROKER_RESULT_ERROR, ((ObjectNode) rootNode).findValue("errorMessage").asText());
```

Comment:
Sometimes the error message might be helpful.  But looks like each row is an json object now, we cannot use the if condition 'rootNode instanceof ObjectNode',  can we somehow keep the logic of extracting the error message?

Reply:
I have similar feeling with @beinan could we extract error message?

Reply:
@beinan @zhenxiao 
I agree, how about checking if the error rootNode `has` errorMessage filed and column handles doesn't contain the field at the same time. If so, we raise the exception with the errorMessage.

Reply:
sure. go ahead

Reply:
> @beinan @zhenxiao
> I agree, how about checking if the error rootNode `has` errorMessage filed and column handles doesn't contain the field at the same time. If so, we raise the exception with the errorMessage.

sounds good to me, it would be a good solution. Thanks!